### PR TITLE
📃 Prompt Eng Roadmap (Image Prompting) - Improve (2023, Q2 -> 2024, Q4)

### PR DESCRIPTION
**Pull request by iPoetDev**
Update topic / sub topics from May 2023 to Oct 2024

- data: `src/data/roadmaps/prompt-engineering/content/108-image-prompting`

- why: Sub topics were empty of content, with only links.

- modify topics: 
   - add content 
       - 100-style-modifers.md
       - 101-quality boosters.md
       - 102-weighted-terms.md
       - 103-deformed-generations.md
   - add links
       - 102-weighted-terms.md

### CODE REVIEW

**Comment by iPoetDev** [comment]
Context: Hacktober submission.
Part of a series: Update entire Prompt Engineering roadmap.
#first-time-PR in Open Source, via fork

File: src/data/roadmaps/prompt-engineering/content/108-image-prompting/100-style-modifiers.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
 # Style Modifiers
 
+Style modifiers are essential tools in AI image prompting that allow users to specify the style of generated images. These descriptors can include elements such as art styles, materials, techniques, and even moods or historical periods. By combining multiple style modifiers, users can achieve highly specific and creative outputs. Recent advancements have expanded the categories of style modifiers and improved AI models' ability to interpret them. Additionally, these modifiers are now supported by a wider range of AI tools, enhancing their applicability in various creative fields.
+
+
 - [@article@Style Modifiers](https://learnprompting.org/docs/image_prompting/style_modifiers)
+
```
Comment (offensive-vk) [inline src/data/roadmaps/prompt-engineering/content/108-image-prompting/100-style-modifiers.md:7]:
```suggestion

Visit the following resources to learn more:

- [@article@Style Modifiers](https://learnprompting.org/docs/image_prompting/style_modifiers)

```

File: src/data/roadmaps/prompt-engineering/content/108-image-prompting/101-quality-boosters.md [hunk 2]
```diff
@@ -1,3 +1,26 @@
...
+• **Style Modifiers**: Adding style-related terms such as "masterpiece" or "art station" can further refine the aesthetic quality of the image, blending artistic styles with technical enhancements.
+
+• **Iterative Feedback and Refinement**: Implementing feedback loops where users can iteratively refine images based on additional inputs or preferences can lead to more personalized and higher-quality results.
+
+• **Experimentation with Prompt Lengths**: Testing various prompt lengths, from concise to detailed, can help determine the most effective approach for different AI tools.
+
+These strategies demonstrate how specific prompt modifications and the use of AI tools can drastically improve the quality of AI-generated images.
+
+
+- [@article@Quality Boosters - learnprompting.org](https://learnprompting.org/docs/image_prompting/quality_boosters)
```
Comment (offensive-vk) [inline src/data/roadmaps/prompt-engineering/content/108-image-prompting/101-quality-boosters.md:26]:
```suggestion
Visit the following resources to learn more:

- [@article@Quality Boosters - learnprompting.org](https://learnprompting.org/docs/image_prompting/quality_boosters)
```

File: src/data/roadmaps/prompt-engineering/content/108-image-prompting/102-weighted-terms.md [hunk 3]
```diff
@@ -1,3 +1,27 @@
...
+- **Emphasizing Elements**: If you want an image with a prominent sunset, you might use sunset (bright colors:1.2).
+- **Excluding Elements**: To avoid trees in a landscape, you could use landscape | tree:-10, ensuring trees are not included in the generated image
+
+### Considerations
+
+- **Impact of Term Placement**: Words at the beginning of a prompt generally have more influence than those at the end.
+- **Iterative Fine-tuning**: Adjusting weights is often an iterative process, requiring multiple attempts to achieve the desired result.
+- **Limitations**: Overemphasis on certain terms may limit creativity and diversity in generated images.
+
+---
```
Comment (offensive-vk) [inline src/data/roadmaps/prompt-engineering/content/108-image-prompting/102-weighted-terms.md:23]:
```suggestion
Visit the following resources to learn more:
```

File: src/data/roadmaps/prompt-engineering/content/108-image-prompting/103-deformed-generations.md [hunk 4]
```diff
@@ -1,3 +1,20 @@
...
+- **Iterative Process**: Achieving the desired image may require multiple attempts, as the technique is not always consistent
+- **Feedback Loops**: Implement feedback mechanisms where users can rate or correct generated images, allowing the model to learn from its mistakes and improve over time.
+- **Regular Updates**: Keep models updated with the latest training data and techniques to ensure they incorporate recent advancements in AI image generation.
+- **Post-Processing Techniques**: Utilize software tools for post-processing, such as image editing applications, to correct minor deformities and enhance overall image quality.
+
+These strategies are essential for anyone working with AI-generated imagery to ensure outputs meet desired standards of quality and realism. As generative models continue to evolve, the need for such techniques is expected to decrease.
+
+Overall, while current models may still struggle with certain deformations, employing negative prompts effectively can significantly enhance image quality.
+
+- [@article@Fix Deformed Generations](https://learnprompting.org/docs/image_prompting/fix_deformed_generations)
```
Comment (offensive-vk) [inline src/data/roadmaps/prompt-engineering/content/108-image-prompting/103-deformed-generations.md:20]:
```suggestion
Visit the following resources to learn more:

- [@article@Fix Deformed Generations](https://learnprompting.org/docs/image_prompting/fix_deformed_generations)
```

**Review by offensive-vk — COMMENTED** [review]
Your first pr is really good, but first, please have a good look of our contributing guidelines here. https://github.com/kamranahmedse/developer-roadmap/blob/master/contributing.md#adding-content

Thankyou for your contributions. 🙌

**Comment by dansholds** [comment]
> Your first pr is really good, but first, please have a good look of our contributing guidelines here. https://github.com/kamranahmedse/developer-roadmap/blob/master/contributing.md#adding-content
> 
> Thankyou for your contributions. 🙌

@offensive-vk please stop reviewing other peoples PR's and saying "our contributing guidelines", this gives the impression that you're a **core team member** and can cause confusion in other contributors.

@iPoetDev Our contribution guides state a single and concise paragraph for the content and the real power comes from the links that are added, the following is an example of a fully loaded topic:

```
# Redis

Redis is an open-source, in-memory data structure store known for its speed and versatility. It supports various data types, including strings, lists, sets, hashes, and sorted sets, and provides functionalities such as caching, session management, real-time analytics, and message brokering. Redis operates as a key-value store, allowing for rapid read and write operations, and is often used to enhance performance and scalability in applications. It supports persistence options to save data to disk, replication for high availability, and clustering for horizontal scaling. Redis is widely used for scenarios requiring low-latency access to data and high-throughput performance.

Learn more from the following resources:

[@official@Link 1](https:/google.com)
[@article@Link 2](https:/google.com)
[@article@Link 3](https:/google.com)
[@course@Link 4](https:/google.com)
[@course@Link 5](https:/google.com)
[@video@Link 6](https:/google.com)
[@video@Link 7](https:/google.com)
[@video@Link 8](https:/google.com)
```

**Review by dansholds — CHANGES_REQUESTED** [review]
Requesting changes as per my above comment.

**Comment by iPoetDev** [comment]
@dansholds 

Understood about a single paragraph intent (as intended in Q4 2024). However, in push back, the content from May 2023 as not info designed in this way.

My stratagem was to extend the current content and fill in the gaps. (Expand).

I get the approach to single paragraph (as it can reduce maintenance and content variability/revision load). 

It is a thing of several days' effort that I have expanded and edits for 54 topics, and was in the process of refining the content in VSCode before commit.

So I will seek some more interaction with you via other channels to get a good formulation/balance between efforts given and effort desired.

I can do the links, BTW>

**Comment by dansholds** [comment]
@iPoetDev, Thanks Charles, this is much more in line 👍🏻

**Review by dansholds — APPROVED** [review]

**Comment by iPoetDev** [comment]
Thank you very much. My first every OS merge accept.

**Comment by iPoetDev** [comment]
Light Admin: Wrong commit message on [8d7672e](https://github.com/kamranahmedse/developer-roadmap/pull/7571/commits/8d7672e)

```

📃 docs, data (Image Prompting) Update Topic/Sub Topics - In Place Edits.

- intent: Update topic from May 2023 to Oct 2024

- data: src/data/roadmaps/prompt-engineering/content/
   - 📂108-image-prompting

- add: New content 
102-weighted-terms.md,100-style-modifers.md,101-quality-boosters.md,103-defomed-generations.md
---

Co-authored-by: [@iPoetDev](https://github.com/iPoetDev) <ipoetdev-github-no-reply@outlook.com>
```

**Pull request merged**